src/EnergyPlus/DXCoils.cc

```diff
@@ -7988,48 +7907,68 @@ namespace DXCoils {
                 if (DXCoil(DXCoilNum).MSRatedTotCap(Mode) == AutoSize) {
                     IsAutoSize = true;
                 }
+
                 if (Mode == DXCoil(DXCoilNum).NumOfSpeeds) {
-                    // Heating capacity is assumed to be equal to the cooling capacity
-                    if (DXCoil(DXCoilNum).CompanionUpstreamDXCoil > 0) {
-                        NumOfSpeedCompanion = DXCoil(DXCoil(DXCoilNum).CompanionUpstreamDXCoil).NumOfSpeeds;
-                        MSRatedTotCapDes = DXCoil(DXCoil(DXCoilNum).CompanionUpstreamDXCoil).MSRatedTotCap(NumOfSpeedCompanion);
-                    } else {
-                        MSRatedTotCapDes = DXCoil(DXCoilNum).RatedTotCap(1); // sized above
+
+                    PrintFlag = true;
+                    DataIsDXCoil = true;
+                    CompName = DXCoil(DXCoilNum).Name;
+                    CompType = DXCoil(DXCoilNum).DXCoilType;
+                    SizingMethod = HeatingCapacitySizing;
+                    FieldNum = 10 + (Mode - 1) * 5;
+                    SizingString = DXCoilNumericFields(DXCoilNum).PerfMode(1).FieldNames(FieldNum) + " [W]";
+                    DataTotCapCurveIndex = DXCoil(DXCoilNum).MSCCapFTemp(Mode);
+                    if (IsAutoSize || !HardSizeNoDesRun) {
+                        // Heating capacity is assumed to be equal to the cooling capacity
+                        SizingMethod = AutoCalculateSizing;
+                        DataFractionUsedForSizing = 1.0;
+                        if (DXCoil(DXCoilNum).CompanionUpstreamDXCoil > 0) {
+                            NumOfSpeedCompanion = DXCoil(DXCoil(DXCoilNum).CompanionUpstreamDXCoil).NumOfSpeeds;
+                            DataConstantUsedForSizing = DXCoil(DXCoil(DXCoilNum).CompanionUpstreamDXCoil).MSRatedTotCap(NumOfSpeedCompanion);
```

Comment:
This code is inside the if (Mode == DXCoil(DXCoilNum).NumOfSpeeds) { section. You should not use DataConstantUsedForSizing and DataFractionUsedForSizing on the highest speed. The reason is that you will not get the Design vs User capacities if those are different by 10%. For the lower speeds using the Data* variables is OK (lines 7940-7958 below). For sizing air flows you did it correctly (lines 7822-7849 above).